backend/src/test/java/io/github/projektalmanac/amoxtli/backend/service/BookServiceTest.java

```diff
@@ -0,0 +1,172 @@
+import org.springframework.mail.javamail.JavaMailSender;
+
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hibernate.validator.internal.util.Contracts.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private GoogleBookService googleBookService;
+
+    @Mock
+    private BookRepository bookRepository;
+
+    @InjectMocks
+    private BookService bookService;
+
+
+    private User user;
+    private User user1;
+    private Book book1;
+    private Volume.VolumeInfo volumeInfo;
+    private Volume.VolumeInfo.ImageLinks imageLinks;
+    private LibroRegistradoDto libroRegistradoDto, libroRegistradoDto2;
+
+    @BeforeEach
+    void setUp() {
+
+        user = new User();
+        user.setId(1);
+        user.setName("Eduardo");
+        user.setLastName("Castro");
+        user.setEmail("ecastro@gmail.com");
+        user.setPhone("58587599");
+        user.setVerifiedEmail(false);
+
+        book1 = new Book();
+        book1.setId(1);
+        book1.setIsbn("1111111111");
+        book1.setDescription("Casi nuevo el libro");
+
+        volumeInfo = new Volume.VolumeInfo();
+        imageLinks = new Volume.VolumeInfo.ImageLinks();
+
+        volumeInfo.setAuthors(Arrays.asList("Autor1", "Autor2", "Autor3"));
+        volumeInfo.setTitle("Titulo 1");
+        imageLinks.setMedium("http://books.google.com/books/content?id=O9ztAAAAMAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api");
+        volumeInfo.setImageLinks(imageLinks);
+        volumeInfo.setCategories(Arrays.asList("Categoria1", "Categoria2"));
+        volumeInfo.setPublisher("Editorial 1");
+        volumeInfo.setDescription("Sinopsis del libro");
+        volumeInfo.setLanguage("Español");
+        volumeInfo.setPublishedDate("1990");
+
+        libroRegistradoDto = new LibroRegistradoDto();
+        libroRegistradoDto.setId(1);
+        libroRegistradoDto.setIsbn("1111111111");
+        libroRegistradoDto.setDescripcion("Casi nuevo el libro");
+
+        libroRegistradoDto2 = new LibroRegistradoDto();
+        libroRegistradoDto2.setIsbn("ashwydr");
+
+        user1 = new User();
+        user1.setId(1);
+        user1.setName("Eduardo");
+        user1.setLastName("Castro");
+        user1.setEmail("ecastro@gmail.com");
+        user1.setPhone("58587599");
+        user1.setVerifiedEmail(false);
+        user1.addBook(book1);
+
+    }
+
+    @Test
+    void getDetallesLibro() {
+
+        //Caso 1: El ISBN no es proporcionado o está vacío
+        final String ISBN = null;
+        assertThrows(ResourceNotFoundException.class, () -> {
+            bookService.getDetallesLibro(ISBN);
+        });
```

Comment:
El comentario dice que pruebas lo que pasa si el ISBN es null o vacío, pero sólo estás probando qué pasa cuando es null, no cuando es vacío.

Lo ideal sería que tuvieras casos de prueba para ambos, pero si quieres puedes simplemente editar el comentario para que corresponda realmente con lo que estás haciendo.

```suggestion
        //Caso 1: El ISBN no es proporcionado
        final String ISBN = null;
        assertThrows(ResourceNotFoundException.class, () -> {
            bookService.getDetallesLibro(ISBN);
        });
```